pkg/controller/utils/secrets.go

```diff
@@ -0,0 +1,36 @@
+package utils
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+
+	k8sDeepHash "github.com/openshift/hive/contrib/pkg/hash"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// LoadSecretData loads a given secret key and returns it's data as a string.
+func LoadSecretData(c client.Client, secretName, namespace, dataKey string) (string, error) {
+	s := &corev1.Secret{}
+	err := c.Get(context.TODO(), types.NamespacedName{Name: secretName, Namespace: namespace}, s)
+	if err != nil {
+		return "", err
+	}
+	retStr, ok := s.Data[dataKey]
```

Comment:
You need to decode this from base64.

Reply:
Actually this function was just moved from another file to this file as I did some restructuring. I can do that change in a different commit as it is a change not required by `global pull secret` feature. @dgoodwin WDYT

Reply:
Is the code working right now? I'd expect you need to base64 decode this at some point at least, when you do the json merge. But yes probably shouldn't happen here as this function is used elsewhere in it's current form.

Reply:
This is not required as my testing as k8s is taking care of encoding and decoding. Not sure if I need to do anything here.